addons/mrp/models/product.py

```diff
@@ -30,6 +30,34 @@ def _compute_used_in_bom_count(self):
             template.used_in_bom_count = self.env['mrp.bom'].search_count(
                 [('bom_line_ids.product_tmpl_id', '=', template.id)])
 
+    def _compute_quantities_dict(self):
+        bom_kits = self.env['mrp.bom'].search([('type', '=', 'phantom'), ('product_tmpl_id', 'in', self.ids)])
+        dict_single_bom_templates = {x.product_tmpl_id: x for x in bom_kits}
+        # Check if template only have 1 type of bom
+        for b in bom_kits:
+            if b.product_tmpl_id in dict_single_bom_templates and dict_single_bom_templates[b.product_tmpl_id] != b:
+                del dict_single_bom_templates[b.product_tmpl_id]
+        single_bom_templates = sum(dict_single_bom_templates, start=self.env['product.template'])
+        prod_available = super(ProductTemplate, self - single_bom_templates)._compute_quantities_dict()
+        variants_available = single_bom_templates.mapped('product_variant_ids')._product_available()
```

Comment:
don't need mapped here

Reply:
Indeed, i mapped to product_variant_id instead to only get the first